ksql-functional-tests/src/test/resources/query-validation-tests/max-group-by.json

```diff
@@ -136,6 +136,49 @@
         {"topic": "S2", "key": 0, "value": {"ID":  0, "MAX":  null}},
         {"topic": "S2", "key": 0, "value": {"ID":  0, "MAX":  1.0}}
       ]
+    },
+    {
+      "name": "max decimal group by",
+      "statements": [
+        "CREATE STREAM TEST (ID bigint, NAME varchar, VALUE decimal(4, 2)) WITH (kafka_topic='test_topic', value_format='DELIMITED', key='ID');",
+        "CREATE TABLE S2 as SELECT id, max(value) FROM test group by id;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": "0,100,00.00"},
+        {"topic": "test_topic", "key": 0, "value": "0,100,05.10"},
+        {"topic": "test_topic", "key": 100, "value": "100,100,10.10"},
+        {"topic": "test_topic", "key": 100, "value": "100,100,06.40"},
+        {"topic": "test_topic", "key": 100, "value": "100,100,30.80"},
+        {"topic": "test_topic", "key": 0, "value": "0,zero,20.99"},
+        {"topic": "test_topic", "key": 0, "value": "0,100,10.11"}
+
+      ],
+      "outputs": [
+        {"topic": "S2", "key": 0, "value": "0,00.00"},
+        {"topic": "S2", "key": 0, "value": "0,05.10"},
+        {"topic": "S2", "key": 100, "value": "100,10.10"},
+        {"topic": "S2", "key": 100, "value": "100,10.10"},
+        {"topic": "S2", "key": 100, "value": "100,30.80"},
+        {"topic": "S2", "key": 0, "value": "0,20.99"},
+        {"topic": "S2", "key": 0, "value": "0,20.99"}
+      ]
+    },
+    {
+      "name": "max decimal by with nulls",
```

Comment:
I'd personally roll this into the test above, so the one test covers null and non-null.

Reply:
I'll have a minor follow-up to do this because it's just kinda annoying and I want to get this in :)